Python/ceval.c

```diff
@@ -6127,6 +6243,200 @@ do_raise(PyThreadState *tstate, PyObject *exc, PyObject *cause)
     return 0;
 }
 
+/* Logic for matching an exception in an except* clause (too
+   complicated for inlining).
+*/
+
+static int
+exception_group_match(
+    PyObject *exc_type, PyObject* exc_value, PyObject *match_type,
+    PyObject **match, PyObject **rest)
+{
+    if (exc_type == Py_None) {
+        assert(exc_value == Py_None);
+        *match = Py_NewRef(Py_None);
+        *rest = Py_NewRef(Py_None);
+        return 0;
+    }
+    assert(PyExceptionClass_Check(exc_type));
+    assert(PyExceptionInstance_Check(exc_value));
+
+    int res = PyErr_GivenExceptionMatches(exc_type, match_type);
+    if (res == -1) {
+        return -1;
+    }
+    if (res > 0) {
```

Comment:
`PyErr_GivenExceptionMatches` returns true/1 or false/0.

```suggestion
    if (res) {
```